File: 21551-naveen-siva-kumar/week7/11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size()-1;
        
        int ans = INT_MIN;//,min=INT_MIN;
        
        while(left<right){
            
            ans = std::max(ans,std::min(height[right],height[left])*(right-left));
            if(height[right]>height[left]) { left+=1;}
            else if(height[right]<height[left]){ right -=1;}
            else{ left+=1;right-=1;}
        }
        return ans;
    }
    
};

File: 21557 - Mukesh J/Week-7/11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n= height.length;
        int i=0, j=n-1,volume=0,max=0;
        while(i<j)
        {
            volume= (j-i)*Math.min(height[j],height[i]);
            if(height[j]<height[i])
                j--;
            else if(height[j]>height[i])
                i++;
            else
            {
                i++;
                j--;
            }
            if(max<volume)
                max=volume;
        }
        return max;
    }
}